### ISSUE TITLE
Extend documentation for contributing

### ISSUE THREAD

**Issue by isabellachen** [Automattic/jetpack#13180]
#### Is your feature request related to a problem? Please describe.

I'm interested in contributing to JetPack (specifically I have a [problem with the lazy-images feature](https://github.com/Automattic/jetpack/issues/9595#issuecomment-460327062)) and would like to see if I can find out what's going on.

I followed the instructions in the readme and got my environment up and running with the docker container. I added a console.log in the lazy-images.js file and `yarn build` followed by `yarn watch` and I saw the message in the console. When I tried to change the message in the console.log, the change was not reflected until I exited the watch process and ran `yarn watch` again.

#### Describe the solution you'd like

I'm wondering if there is a straightforward workflow to working with the js files in the module folder? Basically I would like to be able to make a change in the js file and see it reflected without having to exit the watch process each time.

Futhurmore, I would like to know what the launch.json configuration should be for VSCode if I am using the docker container. Usually when using Flywheel's local, I have a pathMappings property which allows me to put a breakpoint anywhere in a php file, but I don't know what the mapping should be in this case...
```
"pathMappings": {
				"/app/public": "${workspaceRoot}"
			}
```

#### Describe alternatives you've considered

I thought of adding some more gulp.tasks to the gulp file to watch for changes to the js files in the module folder, just like how the watcher watches the jsx and scss files, but I'm not sure if this would work either. And it seems like an obvious solution to me, so if its not already a task in the gulpfile, I guess there must be some misunderstanding on my part on how best to do this,

It would be great if someone could point me in the right direction for how best to start development with the js in the modules because following the documentation didn't get it to work for me. Unless the solution is to simply debug directly in chrome by putting breakpoints directly into the code in the sources tab?

**Comment by jeherve** [comment]
Thanks for the questions. All those things could definitely benefit from better documentation.

> I'm wondering if there is a straightforward workflow to working with the js files in the module folder? Basically I would like to be able to make a change in the js file and see it reflected without having to exit the watch process each time.

If you set `define( 'SCRIPT_DEBUG', true );` in your site's `wp-config.php` file, you'll be able to work with the unminified version of the module files (`modules/lazy-images/js/lazy-images.js` in your case), and thus will see your changes right away.

> I would like to know what the launch.json configuration should be for VSCode if I am using the docker container. Usually when using Flywheel's local, I have a pathMappings property which allows me to put a breakpoint anywhere in a php file, but I don't know what the mapping should be in this case

We currently only have documentation for Atom and PHPStorm:
https://github.com/Automattic/jetpack/blob/53c51ff77c8899ee2ec86b7fc13fb6e804d71c27/docker/README.md#debugging-php-with-xdebug

Documentation for VS Code would definitely be nice.

**Comment by isabellachen** [comment]
@jeherve Thanks for the reply! `define( 'SCRIPT_DEBUG', true );` worked for me. Do you want me to update the development-environment.md to include this?

**Comment by jeherve** [comment]
Yes, I think that would be a nice addition!

**Comment by isabellachen** [comment]
@jeherve I updated the readme but have problems pushing to the repo. 
`ERROR: Permission to Automattic/jetpack.git denied to isabellachen.
fatal: Could not read from remote repository.`

I [cloned the repo with ssh ](https://github.com/Automattic/jetpack/blob/master/docs/development-environment.md#start-development) according to the docs and when I [run the check for the environment ](https://github.com/Automattic/jetpack/blob/master/docs/development-environment.md#script-for-checking-if-your-environment-is-ready-for-contributing-to-jetpack) the tools for contributing all pass. My ssh connection is working too. Am I missing something?

![image](https://user-images.githubusercontent.com/9280665/62486871-9e5cda80-b7c0-11e9-8597-f3cb6d36429d.png)

![image](https://user-images.githubusercontent.com/9280665/62487177-7a4dc900-b7c1-11e9-8cb5-765f0c85c12b.png)

**Comment by jeherve** [comment]
@isabellachen That's indeed to be expected, as you are not currently allowed to push to this repository. Instead, I would recommend that you clone this repository, and work there:

1. At the top right of this page, click on the "Fork" button.
2. Once the repo has been forked, you will have your own copy of Jetpack, at https://github.com/isabellachen/jetpack
3. Back in your terminal, in your working directory for Jetpack, add `git remote add isabellachen git@github.com:isabellachen/jetpack.git`
4. You can now push the branch you've created to your remote fork of Jetpack: `git push -u isabellachen update/readme-development-env`
5. Once you've done that, you should get a banner from GitHub inviting you to open a Pull Request.

**Comment by isabellachen** [comment]
Added documentation for VSCode: https://github.com/Automattic/jetpack/pull/13211#issuecomment-520265509

**Comment by jeherve** [comment]
Fixed in #13211
